static/js/components/inputs/SelectField.js

```diff
@@ -72,34 +63,46 @@ class SelectField extends React.Component {
     return `select-field ${classify(className)} ${classify(label)}`;
   };
 
+  formattedId = (): string => {
+    const { id, label } = this.props;
+    return id ? id : _.uniqueId(classify(label));
+  }
+
   selectClassName = (): string => {
     const { errors, keySet, topMenu } = this.props;
     return `${validationErrorSelector(errors, keySet)} ${topMenu ? 'menu-outer-top' : ''}`;
   };
 
-  render() {
-    const { errors, keySet, profile, label } = this.props;
-    const { id } = this.state;
-    const select = (
-      <VirtualizedSelect
-        value={_.get(profile, keySet, "")}
-        className={this.selectClassName()}
-        onChange={this.onChange}
-        onBlur={this.onBlur}
-        {...this.props}
-      />
-    );
-    let labelledSelect;
-    if (label) {
-      labelledSelect = <label className="react-select-label">
-        {label}{select}
-      </label>;
-    } else {
-      labelledSelect = select;
+  label = (label: string, element: React$Element<*>) => (
+    <label className="react-select-label">
+      { label }
+      { element }
+    </label>
+  );
+
+  renderSelect = (): React$Element<*> => {
+    const { keySet, profile, allowCreate, options } = this.props;
+
+    let value = _.get(profile, keySet);
+    if ( allowCreate && value && !options.find(option => option.value === value) ) {
+      options.push({ value: value, label: value });
```

Comment:
It seems to work fine when I use `.push`, do you mean we shouldn't alter props? My thought in using `.push` was that I want the option to persist between re-renders, so that if you enter a 'created' option and then change the value, you can change your selection and your created value will still be there. If I use `concat` we don't get that behavior, so I would need to implement something more complicated, I think, to keep track of new `Option`s which had been added.